src/app/models/submission.server.model.ts

```diff
@@ -430,13 +436,196 @@ EncryptSubmissionSchema.statics.findEncryptedSubmissionById = function (
+  },
+  encryptedSubmissionSecretKey: {
+    type: String,
+    trim: true,
+    required: true,
+  },
+  encryptedContent: {
+    type: String,
+    trim: true,
+    required: true,
+  },
+  verifiedContent: {
+    type: String,
+    trim: true,
+  },
+  attachmentMetadata: {
+    type: Map,
+    of: String,
+  },
+  version: {
+    type: Number,
+    required: true,
+  },
+})
+
+MultirespondentSubmissionSchema.statics.findSingleMetadata = function (
+  formId: string,
+  submissionId: string,
+): Promise<SubmissionMetadata | null> {
+  const pageResults: Promise<MetadataAggregateResult[]> = this.aggregate([
+    {
+      $match: {
+        submissionType: SubmissionType.Multirespondent,
+        form: mongoose.Types.ObjectId(formId),
+        _id: mongoose.Types.ObjectId(submissionId),
+      },
+    },
+    { $limit: 1 },
+  ]).exec()
+
+  return Promise.resolve(pageResults).then((results) => {
+    if (!results || results.length <= 0) {
+      return null
+    }
+
+    const result = results[0]
+
+    // Build submissionMetadata object.
+    const metadata = buildSubmissionMetadata(result, 1)
+
+    return metadata
+  })
+}
+
+MultirespondentSubmissionSchema.statics.findAllMetadataByFormId = function (
+  formId: string,
+  {
+    page = 1,
+    pageSize = 10,
+  }: {
+    page?: number
+    pageSize?: number
+  } = {},
+): Promise<{
+  metadata: SubmissionMetadata[]
+  count: number
+}> {
+  const numToSkip = (page - 1) * pageSize
+  // return documents within the page
+  const pageResults: Promise<MetadataAggregateResult[]> = this.aggregate([
+    { $match: { form: mongoose.Types.ObjectId(formId) } },
+    { $sort: { created: -1 } },
+    { $skip: numToSkip },
+    { $limit: pageSize },
+    {
+      $project: {
+        _id: 1,
+        created: 1,
+      },
+    },
+  ]).exec()
+
+  const count =
+    this.countDocuments({
+      form: mongoose.Types.ObjectId(formId),
+      submissionType: SubmissionType.Multirespondent,
+    }).exec() ?? 0
+
+  return Promise.all([pageResults, count]).then(([results, count]) => {
+    let currentNumber = count - numToSkip
+
+    const metadata = results.map((result) => {
+      const paymentMeta = result.payments?.[0]
+      const metadataEntry = buildSubmissionMetadata(
+        result,
+        currentNumber,
+        paymentMeta,
+      )
+
+      currentNumber--
+      return metadataEntry
+    })
+
+    return {
+      metadata,
+      count,
+    }
+  })
+}
+
+MultirespondentSubmissionSchema.statics.getSubmissionCursorByFormId = function (
+  formId: string,
+  dateRange: {
+    startDate?: string
+    endDate?: string
+  } = {},
+): QueryCursor<MultirespondentSubmissionCursorData> {
+  const streamQuery = {
+    form: formId,
+    ...createQueryWithDateParam(dateRange?.startDate, dateRange?.endDate),
+  }
+  return (
+    this.find(streamQuery)
+      .select({
+        submissionType: 1,
+        form_fields: 1,
+        form_logics: 1,
+        encryptedSubmissionSecretKey: 1,
```

Comment:
Here we don't need it. The submission public key is only required when the future respondents 2+ enter their secret key, to validate that the key is correct. Since this cursor function is only used when the admin exports as CSV, so it's not required. In contrast, it is required for `findEncryptedSubmissionById` since that function retrieves the encrypted response for the respondent 2+.